db/table_cache.cc

```diff
@@ -197,4 +201,144 @@ TableCache::GetStatisticValue(
 
 }   // TableCache::GetStatisticValue
 
+
+/**
+ * Riak specific routine to push list of open files to disk
+ */
+void
+TableCache::SaveOpenFileList()
+{
+    Status s;
+    std::string cow_name;
+    WritableFile * cow_file;
+    log::Writer * cow_log;
+
+    cow_name=CowFileName(dbname_);
+    s = env_->NewWritableFile(cow_name, &cow_file, 4*1024L);
+    if (s.ok())
+    {
+        std::string buffer;
```

Comment:
I recommend setting the initial capacity of the buffer variable (say, to 4KB) to avoid a bunch of re-allocs as cache file records are added to the std::string.